community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/UsingAcceptanceTest.scala

```diff
@@ -225,4 +225,23 @@ class UsingAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSup
     intercept[SyntaxException](
       executeWithAllPlanners("MATCH n-->() USING SCAN n:Person return n"))
   }
+
+  test("when failing to support all hints we should provide an understandable error message") {
+    // GIVEN
+    graph.createIndex("LocTag", "id")
+
+    // WHEN
+    val query = """CYPHER planner=greedy MATCH (t1:LocTag {id:1642})-[:Child*0..]->(:LocTag)
+                  |     <-[:Tagged]-(s1:Startup)<-[r1:Role]-(u:User)
+                  |     -[r2:Role]->(s2:Startup)-[:Tagged]->(:LocTag)
+                  |     <-[:Child*0..]-(t2:LocTag {id:1642})
```

Comment:
This is and old test moved over from UsingInAcceptanceTest, as mentioned above these two were exact copies saved from this very test.

The reason why this test is so complicated is that it was a bug that was really hard to reproduce, but it is an old test not added in this PR.